static/app/views/performance/landing/widgets/transforms/transformMetricsToArea.tsx

```diff
@@ -0,0 +1,78 @@
+import mean from 'lodash/mean';
+import moment from 'moment';
+
+import {getParams} from 'sentry/components/organizations/globalSelectionHeader/getParams';
+import {defined} from 'sentry/utils';
+import {axisLabelFormatter} from 'sentry/utils/discover/charts';
+import {aggregateOutputType} from 'sentry/utils/discover/fields';
+import {MetricsRequestRenderProps} from 'sentry/utils/metrics/metricsRequest';
+
+import {QueryDefinitionWithKey, WidgetDataConstraint, WidgetPropUnion} from '../types';
+
+export function transformMetricsToArea<T extends WidgetDataConstraint>(
+  widgetProps: WidgetPropUnion<T>,
+  results: MetricsRequestRenderProps,
+  _: QueryDefinitionWithKey<T>
+) {
+  const {start, end, utc, interval, statsPeriod} = getParams(widgetProps.location.query);
+
+  const {errored, loading, reloading, response, responsePrevious} = results;
+
+  const metricsField = widgetProps.fields[0];
+
+  const data =
+    response?.groups.map(group => {
+      return {
+        seriesName: metricsField,
+        data: response.intervals.map((intervalValue, intervalIndex) => {
+          return {
+            name: moment(intervalValue).valueOf(),
+            value: group.series[metricsField][intervalIndex],
+          };
+        }),
+      };
+    }) ?? [];
+
+  const dataMean = data.map(series => {
+    const meanData = mean(series.data.map(({value}) => value));
+    return {
+      mean: meanData,
+      outputType: aggregateOutputType(series.seriesName),
+      label: axisLabelFormatter(meanData, series.seriesName),
```

Comment:
We need to make sure that the output of these functions is correct for all fields.
Not sure if our naming is 1:1 with all transaction fields (tpm and failure_rate come to my mind).

Reply:
sure... I might have to update it when implementing tpm

Reply:
it works for tpm https://github.com/getsentry/sentry/blob/efc144f6069dcf510122fddb5c3906a9bd642eeb/static/app/utils/discover/fields.tsx#L117